lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_server
+
+short_description: Create and manage cloud servers on the Hetzner Cloud
+
+version_added: "2.8"
+
+description:
+    - "Create and manage cloud servers on the Hetzner Cloud."
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    id:
+        description:
+            - The Server ID
+            - Only required if no server name is given
+        type: int
+    name:
+        description:
+            - The Server Name
+            - Only required if no server id is given or a server does not exists
+        type: str
+    server_type:
+        description:
+            - The Server Type
+            - Required if server does not exists
+        type: str
+    ssh_keys:
+        description:
+            - List of SSH Keys Names
+        type: list
```

Comment:
ansible has some magic and allows this for type list

~~~yaml
ssh_keys: mykey
~~~

which is identical to 

~~~yaml
ssh_keys: [ mykey ]
~~~

therefore I would put an `aliases: [ ssh_key ]`here and in the arg_spec to make it more conventient: 

~~~yaml
ssh_key: mykey
~~~

Reply:
I disagree, if we end up with people using `ssh_key: my_key` in their playbooks, people may assume only a single key can be added. Keeping only the plural `ssh_keys` form will make clear more than one SSH key is possible, even if only one is provided.

So adding the singular form for convenience introduces potential confusion and inconsistencies between playbooks and users. It's always better to keep the number of aliases to a minimum and only add them for backward compatibility or to add consistency (e.g. using `name` or `path` for abstraction).

Besides it would mean we do this for every plural parameter name (usually lists) which is definitely not true.

Reply:
cc acozine gundalow Should we clarify this in the module development docs ?

Reply:
it is a user interface, if users don't read docs it can't be our fault. anyways. not a blocker (that is why I used "would")

Reply:
@resmo But if adding something for convenience can cause confusion, it may not be in our best interest. Good user interface design should not require more documentation, it should make documentation less important. The ultimate goal would be to not need any documentation because things speak for themselves, and I think that is what `ssh_keys` does in this case.

Reply:
This was discussed in the ansible-docs meeting and we were in general agreement with @dagweirs here, esp clarifying when to use an alias in modules. We'll get that added to developer documentation in a separate PR.